subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc

```diff
@@ -1048,3 +1048,40 @@ Alternatively, for Kotlin implementations, the Kotlin compiler can be configured
 include::sample[dir="snippets/configurationCache/samConversion/groovy",files="build.gradle[tags=configure]"]
 include::sample[dir="snippets/configurationCache/samConversion/kotlin",files="build.gradle.kts[tags=configure]"]
 ====
+
+[[config_cache:not_yet_implemented:accessing_top_level_at_execution]]
+=== Accessing top-level methods and variables of a build script at execution time
+
+Currently, Gradle cannot run a task restored from the configuration cache if a task action calls a method defined directly in the build script.
+
+In case of the Groovy DSL, the action fails when running from the cache with an error message about method not being found.
+In the following snippet, the task `listFiles` uses such method:
+====
+include::sample[dir="snippets/configurationCache/topLevel/groovy",files="build.gradle[tags=not-supported]"]
+====
+
+Running the task the second time when the configuration cache is reused gives an error:
+----
+include::{snippetsPath}/configurationCache/topLevel/tests-groovy/notSupported.out[]
+----
+
+The workaround is to convert the method to the static method of a proper class:
+====
+include::sample[dir="snippets/configurationCache/topLevel/groovy",files="build.gradle[tags=workaround]"]
+====
+
+In case of the Kotlin DSL, the task cannot be stored to the configuration cache at all, because the captured script object reference cannot be serialized.
+For the same reason, it is not possible to reference a top-level variable from the task action too.
+
+====
+include::sample[dir="snippets/configurationCache/topLevel/kotlin",files="build.gradle.kts[tags=not-supported]"]
+====
+
+The workaround follows the same pattern as the Groovy one:
+====
+include::sample[dir="snippets/configurationCache/topLevel/kotlin",files="build.gradle.kts[tags=workaround]"]
+====
+<1> Define the method inside an object
+<2> Define the variable in a smaller scope
+
+See link:{gradle-issues}22879[gradle/gradle#22879].
```

Comment:
```suggestion
For more information about this limitation (and the steps you can take to work around it), see link:{gradle-issues}22879[gradle/gradle#22879].
```
Suggestion: introduce the link.

Reply:
These links are for the users mostly to track when the limitation is lifted, it is a bit misleading to promise more information and workarounds here; the intention is to have all workarounds in the doc. I'm pretty sure the introduction of this link (and similar in other sections) can be reworded to add more clarity, but maybe we should address this separately for all links at once.